pkg/authenticator/authenticator.go

```diff
@@ -215,7 +215,7 @@ func parseIBMCloudCredentials(logger *zap.Logger, data string) (map[string]strin
 // isTimeout ...
 func isTimeout(err error) bool {
 	// If the error message contains "Client.Timeout" substring, return true
-	if strings.Contains(err.Error(), "Client.Timeout") {
+	if strings.Contains(strings.ToLower(err.Error()), "timeout") {
```

Comment:
@GunaKKIBM can you paste the sample error we had covered before and now including the recent error message ?

Reply:
https://github.com/IBM/secret-utils-lib/pull/21#issuecomment-1497116376 - commented @sameshai

Reply:
Can you check the possibility to switch iam endpoint private vs public on the basis of cluster type and use public iam endpoint in case of satellite cluster and IKS/ROKS use private endpoint

Reply:
PR is already merged for always using public endpoint in case of satellite cluster, if the user has not provided any endpoint https://github.com/IBM/secret-utils-lib/pull/22